### ISSUE TITLE
Shims installed by deno on Windows does not work in WSL paths

### ISSUE THREAD

**Issue by felipecrs** [denoland/deno#10135]
```sh-session
> deno install --allow-net --allow-read https://deno.land/std@0.92.0/http/file_server.ts
Download https://deno.land/std@0.92.0/http/file_server.ts
Download https://deno.land/std@0.92.0/path/mod.ts
Download https://deno.land/std@0.92.0/flags/mod.ts
Download https://deno.land/std@0.92.0/_util/assert.ts
Download https://deno.land/std@0.92.0/http/server.ts
Download https://deno.land/std@0.92.0/async/mod.ts
Download https://deno.land/std@0.92.0/http/_io.ts
Download https://deno.land/std@0.92.0/io/bufio.ts
Download https://deno.land/std@0.92.0/path/posix.ts
Download https://deno.land/std@0.92.0/path/common.ts
Download https://deno.land/std@0.92.0/path/glob.ts
Download https://deno.land/std@0.92.0/path/separator.ts
Download https://deno.land/std@0.92.0/path/win32.ts
Download https://deno.land/std@0.92.0/_util/os.ts
Download https://deno.land/std@0.92.0/path/_interface.ts
Download https://deno.land/std@0.92.0/path/_constants.ts
Download https://deno.land/std@0.92.0/path/_util.ts
Download https://deno.land/std@0.92.0/textproto/mod.ts
Download https://deno.land/std@0.92.0/http/http_status.ts
Download https://deno.land/std@0.92.0/async/deferred.ts
Download https://deno.land/std@0.92.0/async/delay.ts
Download https://deno.land/std@0.92.0/async/pool.ts
Download https://deno.land/std@0.92.0/async/mux_async_iterator.ts
Download https://deno.land/std@0.92.0/io/buffer.ts
Download https://deno.land/std@0.92.0/bytes/mod.ts
Download https://deno.land/std@0.92.0/io/util.ts
Check https://deno.land/std@0.92.0/http/file_server.ts
✅ Successfully installed file_server
C:\Users\felip\.deno\bin\file_server.cmd
C:\Users\felip\.deno\bin\file_server (shell)
ℹ️  Add C:\Users\felip\.deno\bin to PATH
    set PATH=%PATH%;C:\Users\felip\.deno\bin
> cd '\\wsl$\ubuntu\home\felipecrs\repos\megatar\'
> C:\Users\felip\.deno\bin\file_server.cmd
'\\wsl$\ubuntu\home\felipecrs\repos\megatar'
CMD.EXE was started with the above path as the current directory.
UNC paths are not supported.  Defaulting to Windows directory.
HTTP server listening on http://0.0.0.0:4507/
```

References https://github.com/npm/cli/issues/1189
References https://github.com/volta-cli/volta/issues/966

**Comment by felipecrs** [comment]
One approach that would solve this issue and #10136 would be by using https://github.com/kiennq/scoop-better-shimexe, as demonstrated here: https://github.com/npm/cli/issues/1189#issuecomment-817209410.

The contents of the `file_server.shim` would be:

```
path = deno.exe
args = run --allow-read --allow-net https://deno.land/std@0.92.0/http/file_server.ts
```

**Comment by nayeemrmn** [comment]
Do powershell scripts support UNC paths? This seems like a cut-and-dry limitation of cmd, so offering something other than cmd is the only solution we should explore here IMO. cc @piscisaureus

**Comment by felipecrs** [comment]
> Do powershell scripts support UNC paths?

Yes.

**Comment by piscisaureus** [comment]
> ```
> > cd '\\wsl$\ubuntu\home\felipecrs\repos\megatar\'
> ```

Using `pushd` instead of `cd` should work.

**Comment by felipecrs** [comment]
> Using `pushd` instead of `cd` should work.
@piscisaureus There isn't any problem with the `cd`.

**Comment by felipecrs** [comment]
@nayeemrmn I have no idea what you are talking about. Replace `cd` with `pushd` where?!

BTW this issue is still valid with Deno 1.44.4. Also worth mentioning that any UNC path is enough to trigger this issue, like browsing network directories.

**Comment by felipecrs** [comment]
Oh, I see now. Sorry. Yeah, that's very tedious indeed lol.

**Comment by felipecrs** [comment]
Note bun created a solution for this issue, they call it `.bunx`:

https://bun.sh/blog/bun-v1.1#bun-run-on-windows

It would be awesome if Deno could leverage it.

**Comment by bartlomieju** [comment]
This is still reproducible with current Deno. `deno install -g` on Windows generates a `.cmd` shim, and `cmd.exe` does not support UNC working directories, so the installed tool starts with its cwd silently switched to `C:\Windows`. `deno.exe` itself handles UNC cwds fine, the problem is purely the `cmd.exe` interpreter in the middle.

The plan to fix this is to replace the `.cmd` shims with real `.exe` shims: `deno.exe` hardlinked (or copied, as a cross-volume fallback) as `<name>.exe`, with the run arguments stored in a sidecar file, dispatching on `argv[0]` at startup. PR #34969 introduces the `argv[0]` dispatch machinery (for a `node` compat shim); once it lands, a follow-up PR will switch `deno install` shims to the same mechanism.

Since the shim then becomes a real console process with no `cmd.exe` in the chain, this will also fix #9873 (the "Terminate batch job (Y/N)?" prompt on Ctrl+C).
